### ISSUE TITLE
Finding Base Addresses

### ISSUE THREAD

**Issue by nikitabuyevich** [jnpl95/Timelapse#13]
Hi. First I want to say how great of a job you did on this trainer! This is very impressive work!

I'm trying to recreate this trainer for v62 but I'm having a lot of trouble finding the Base addresses.

What methods did you use to find so many of these base addresses and offsets?

**Comment by jnpl95** [comment]
I used IDA & Cheat Engine to analyze the executable, tbh finding addresses took most of my time while coding this. The C++ code is easier than actually reversing the functions/finding the addresses/asm to modify.

For finding the pointer addresses, you can use cheat engine, and to find v62 pointers, like for example attack count, you have to scan the values and attack while in game (and then scan again but by increased value by 1). There should be plenty of tutorials online to finding pointers with cheat engine. 

You can look in IDA for the hack addresses and refer to the v95 pdb and look at v62's client and compare the two to find the correct functions & addresses. You can also look online for addresses, maybe someone already posted common gms v62 pointer/hack addresses. Sorry I've never worked on v62 so I can't give you the addresses, otherwise I would. 

So basically look online using google first to see if anyone has posted the pointer address or hack address for v62 that you want to find. If you can't find any, then use cheat engine to scan for the pointers to get the addresses. Use IDA to find functions addresses by comparing to v95 or break&trace using a debugger

**Comment by jnpl95** [comment]
I just searched and found this: https://www.gamekiller.net/threads/maplestory-v62-private-server-hacks.282222/

Idk if it works but you can try downloading the cheat table and testing, I assume that'll be really helpful. Btw this trainer has loads of bugs that I still need to fix, so don't expect greatness hahah 

If you're just trying to hack, just use cheat engine with the packet editor and the cheat table provided in the link above. A trainer with GUI elements is unnecessary if you just use cheat engine and it is not detected.
